ledger/src/database/database_impl.rs

```diff
@@ -20,6 +20,7 @@ pub struct DatabaseImpl<T: TreeVersion> {
     accounts: Vec<Option<T::Account>>,
     pub hashes_matrix: HashesMatrix,
     id_to_addr: HashMap<AccountId, Address>,
+    token_to_account: HashMap<T::TokenId, AccountId>,
```

Comment:
IIRC maintaining this table was costly (thats why @sebastiencs removed it), so we need to add a runtime flag that will toggle it to be a noop when no archive node is used (for later, once everything else is solved I can add that)